src/main/java/harper/utils/Parser.java

```diff
@@ -0,0 +1,174 @@
+import harper.exceptions.HarperInvalidEventException;
+import harper.exceptions.HarperInvalidIndexException;
+import harper.tasks.Deadline;
+import harper.tasks.Event;
+import harper.tasks.Task;
+import harper.tasks.ToDo;
+
+/**
+ * The Parser class is responsible for interpreting user's into corresponding command.
+ * There are different methods that handle different types of input and convert them
+ * into Command object.
+ */
+public class Parser {
+
+    private static final DateTimeFormatter DATE_TIME_FORMATTER = DateTimeFormatter.ofPattern("d/M/yyyy H:mm");
+
+    /**
+     * Parses the command.
+     *
+     * @param command commands.Command entered by user.
+     * @return Different type of commands.Command based on user's input.
+     */
+    public static Command parse(String command) {
+        if (command.equals("bye")) {
+            return new ExitCommand();
+        } else if (command.equals("list")) {
+            return new ListCommand();
+        } else if (command.startsWith("todo ")) {
+            return handleToDo(command);
+        } else if (command.startsWith("deadline ")) {
+            return handleDeadline(command);
+        } else if (command.startsWith("event ")) {
+            return handleEvent(command);
+        } else if (command.startsWith("delete ")) {
+            return handleDelete(command);
+        } else if (command.startsWith("mark ") || command.startsWith("unmark ")) {
+            return handleMark(command);
+        } else if (command.startsWith("find ")) {
+            return handleFind(command);
+        } else {
+            throw new HarperInvalidCommandException();
+        }
+    }
+
+    /**
+     * Parses the todo command.
+     *
+     * @param command Todo command entered by user.
+     * @return commands.AddCommand that adds the todo task into the list.
+     */
+    public static Command handleToDo(String command) {
+        String taskDescription = command.substring("todo".length()).trim();
+        Task newToDo = new ToDo(taskDescription, false);
+        return new AddCommand(newToDo);
+    }
+
+    /**
+     * Parses the deadline command
+     *
+     * @param command task.Deadline command entered by user.
+     * @return commands.AddCommand that adds the deadline task into the list.
+     */
+    public static Command handleDeadline(String command) {
+        String taskDesciptionAndDeadline = command.substring("deadline".length()).trim();
+        String[] parts = taskDesciptionAndDeadline.split("/by", 2);
+        if (parts.length != 2) {
+            throw new HarperInvalidDeadlineException();
+        }
+        String description = parts[0].trim();
+        String deadline = parts[1].trim();
+        if (description.isEmpty() || deadline.isEmpty()) {
+            throw new HarperInvalidDeadlineException();
+        }
+        try {
+            LocalDateTime deadlineFormatted = LocalDateTime.parse(deadline, DATE_TIME_FORMATTER);
```

Comment:
Should this variable be named more sensibly? something like formattedDeadline, similar to formattedEvent and formattedToDo